cobc/typeck.c

```diff
@@ -2420,6 +2420,14 @@ cb_build_identifier (cb_tree x, const int subchk)
 		if (CB_EXCEPTION_ENABLE (COB_EC_BOUND_SUBSCRIPT) && f->odo_level != 0) {
 			for (p = f; p; p = p->children) {
 				if (p->depending && p->depending != cb_error_node
+				    /* Do not check length for implicit access
+				       to a PIC L field (i.e, via enclosing
+				       group), as those disregard the DEPENDING.
+				       However, assuming the filler is never
+				       explicitly accessed (p == f), check is
+				       still done for explicit access to PIC L
+				       field (p->parent == f). */
+				 && (p->parent == f || !p->parent->flag_picture_l)
```

Comment:
Sorry to be such a pain, but as we're shortly before a RC (currently planned end of August) this looks too suspicious to me:

`&& (p->parent == f || !p->parent->flag_picture_l)`
is - for existing code - equal to
`&& (p->parent == f)`
and as the loop always iterates through he grand-children this seems to contradict the loop (now is only active if the direct parent is still `f`).

This is currently the main reason for "not merge ready" - I _think_ this removes the odo-check for grand-children.

Reply:
I think for existing code `(p->parent == f || !p->parent->flag_picture_l)` is equal to true (as no field had `flag_picture_l` true).

Reply:
... that's of course correct and I was wrong.
* p _must_ have a parent here, because its a (grand-)child of `f`
* those never have this attribute set

This leaves us with "the main issue" resolved :-)

Reply:
Nope, I wasn't wrong initially, only when marking that as resolved :-)

I've compiled a bigger codebase with the current nightly which made cobc SIGSEGV here.
Reason: `OCCURS DEPENDING ON` with a non-standard level 01 - goes in here, `f = p`, so `p` has no parent.

I'll add the testcase and the fix to svn in the next hour.

Reply:
Argh… obviously I must also have convinced myself that `p` always had a parent here. Thanks for fixing that then.